### PR TITLE
Advisory command for dnfdaemon

### PR DESCRIPTION
**Pull request by m-blaha**
This PR implements Advisory interface for dnf5daemon-server. Also `dnf5daemon-client advisory list` and `dnf5daemon-client advisory info` commands were implemented.

Tests: https://github.com/rpm-software-management/ci-dnf-stack/pull/1356

Resolves: https://github.com/rpm-software-management/dnf5/issues/451
Resolves: https://bugzilla.redhat.com/show_bug.cgi?id=2184329

### CODE REVIEW

**Comment by Conan-Kudo** [comment]
This will be needed to enable moving dnfdragora from the old dnfdaemon to this one.

**Review by kontura — COMMENTED** [review]
I have just a few minor notes.

File: include/libdnf5-cli/output/advisorylist.hpp [hunk 1]
```diff
@@ -23,12 +23,56 @@ along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
 
 #include <libdnf5/advisory/advisory_package.hpp>
 #include <libdnf5/advisory/advisory_reference.hpp>
+#include <libsmartcols/libsmartcols.h>
 
 namespace libdnf5::cli::output {
 
+// advisory list table columns
+enum { COL_ID, COL_ADVISORY_TYPE, COL_ADVISORY_SEVERITY, COL_ADVISORY_PACKAGE, COL_ADVISORY_BUILDTIME };
```
[the remark this answers is not in the document]
Comment (m-blaha) [reply include/libdnf5-cli/output/advisorylist.hpp]: Hm, items from this enum are passed to smartcols which expects integer. Changing to enum class means that `COL_ADVISORY_TYPE`  needs to be renamed to ugly `static_cast<int>(AdvisoryListColumns::COL_ADVISORY_TYPE)`.
How about moving the enum back to advisorylist.cpp (a new `sort_advisorylist_table()` function is needed). I added a new commit for this for easier review. If you agree with the solution, I can squash the commit.

[the remark this answers is not in the document]
Comment (kontura) [reply include/libdnf5-cli/output/advisorylist.hpp]: It looks good, thank you.

File: dnf5daemon-server/services/advisory/advisory.cpp [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    });
+}
+
+libdnf5::advisory::AdvisoryQuery Advisory::advisory_query_from_options(
+    libdnf5::Base & base, const KeyValueMap & options) {
+    auto opt_type = key_value_map_get<std::vector<std::string>>(options, "type", {});
+    auto opt_severity = key_value_map_get<std::vector<std::string>>(options, "severity", {});
+    auto opt_name = key_value_map_get<std::vector<std::string>>(options, "name", {});
+    auto opt_reference_bz = key_value_map_get<std::vector<std::string>>(options, "reference_bz", {});
+    auto opt_reference_cve = key_value_map_get<std::vector<std::string>>(options, "reference_cve", {});
```
[the remark this answers is not in the document]
Comment (m-blaha) [reply dnf5daemon-server/services/advisory/advisory.cpp]: Makes sense. I've renamed options to plural forms.

File: dnf5daemon-server/dbus/interfaces/org.rpm.dnf.v0.Advisory.xml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+<node>
+<!-- org.rpm.dnf.v0.Advisory:
+   @short_description: Interface to security advisories
+-->
+<interface name="org.rpm.dnf.v0.Advisory">
+    <!--
+        list:
+        @options: an array of key/value pairs
+        @data: array of returned advisories with requested attributes
```
[the remark this answers is not in the document]
Comment (m-blaha) [reply dnf5daemon-server/dbus/interfaces/org.rpm.dnf.v0.Advisory.xml]: Thanks, fixed.

File: include/libdnf5-cli/output/advisorylist.hpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (m-blaha) [reply include/libdnf5-cli/output/advisorylist.hpp]: Thanks, I've squashed the commit to the original one that changed the function to a template.

File: dnf5daemon-server/advisory.cpp [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+            col_module["s"] = mdl.get_stream();
+            col_module["v"] = mdl.get_version();
+            col_module["c"] = mdl.get_context();
+            col_module["a"] = mdl.get_arch();
+            modules.emplace_back(std::move(col_module));
+        }
+
+        KeyValueMap collection;
+        collection["packages"] = std::move(packages);
+        collection["modules"] = std::move(modules);
```
[the remark this answers is not in the document]
Comment (m-blaha) [reply dnf5daemon-server/advisory.cpp:111]: Hm, I cannot see the warning (I'm on F38).

[the remark this answers is not in the document]
Comment (kontura) [reply dnf5daemon-server/advisory.cpp:111]: Sorry, its from `clang-tidy` not a regular `clang` compile.

**Review by kontura — APPROVED** [review]
The build passes locally on rawhide.
The failing rawhide COPR build is unrelated.

**Pull request merged**